Makefile

```diff
@@ -25,125 +41,134 @@ else
 
-# Run unit tests
 .PHONY: test
+test: ## Run unit tests
 test: bin/kubebuilder
 	KUBEBUILDER_ASSETS="$(shell pwd)/bin/kubebuilder/bin" go test ./... -coverprofile cover.out $(ARGS)
 
-# Run end to end tests in a local Kind cluster. We do not clean up after running the tests to
+.PHONY: e2e-kind
+e2e-kind: ## Run end to end tests - creates a new Kind cluster called etcd-e2e
+	go test -v -parallel ${TEST_PARALLEL_E2E} -timeout 20m ./internal/test/e2e \
+			--kind \
+			--repo-root ${CURDIR} \
+			--cleanup=${CLEANUP} \
+			--controller-image=${DOCKER_IMAGE_CONTROLLER} \
+			$(ARGS)
+# We do not clean up after running the tests to
 #  a) speed up the test run time slightly
 #  b) allow debug sessions to be attached to figure out what caused failures
-.PHONY: kind
-kind:
-	go test -parallel ${TEST_PARALLEL_E2E} -timeout 20m ./internal/test/e2e --kind --repo-root ${CURDIR} -v --cleanup=${CLEANUP} $(ARGS)
 
-# Build manager binary
+.PHONY: e2e
+e2e: ## Run the end-to-end tests - uses the current KUBE_CONFIG and context
+e2e:
+	go test -parallel ${TEST_PARALLEL_E2E} -timeout 30m ./internal/test/e2e --current-context --repo-root ${CURDIR} -v $(ARGS)
+
 .PHONY: manager
-manager:
+manager: ## Build manager binary
 	go build -o bin/manager -ldflags="-X 'github.com/improbable-eng/etcd-cluster-operator/version.Version=${VERSION}'" main.go
 
-# Run against the configured Kubernetes cluster in ~/.kube/config
+.PHONY: run
+run: ## Run against the configured Kubernetes cluster in ~/.kube/config
+	DISABLE_WEBHOOKS=1 go run ./main.go
 # Use 'DISABLE_WEBHOOKS=1` to run the controller-manager without the
 # webhook server, and to skip the loading of webhook TLS keys, since these are
 # difficult to set up locally.
-.PHONY: run
-run:
-	DISABLE_WEBHOOKS=1 go run ./main.go
 
-# Install CRDs into a cluster
 .PHONY: install
-install:
+install: ## Install CRDs into a cluster
 	kustomize build config/crd | kubectl apply -f -
 
-# Deploy controller in the configured Kubernetes cluster in ~/.kube/config
-.PHONY: deploy
-deploy:
-	cd config/manager && kustomize edit set image controller=${IMG}
+.PHONY: deploy-cert-manager
+deploy-cert-manager: ## Deploy cert-manager in the configured Kubernetes cluster in ~/.kube/config
+	kubectl apply --validate=false --filename=https://github.com/jetstack/cert-manager/releases/download/v0.11.0/cert-manager.yaml
+	kubectl wait --for=condition=Available --timeout=300s apiservice v1beta1.webhook.cert-manager.io
+
+.PHONY: deploy-controller
+deploy-controller: ## Deploy controller in the configured Kubernetes cluster in ~/.kube/config
+	cd config/manager && kustomize edit set image controller=${DOCKER_IMAGE_CONTROLLER}
 	kustomize build config/default | kubectl apply -f -
+	kubectl --namespace eco-system wait --for=condition=Available --timeout=300s deploy eco-controller-manager
+
+.PHONY: deploy
+deploy: ## Deploy the operator, including dependencies
+deploy: install deploy-cert-manager deploy-controller
```

Comment:
I think the `kustomize build config/default` run in the `deploy-controller` target includes the CRDs, so right now the `install` target is redundant?